### ISSUE TITLE
Case insensitive read name always used regardless of PropertyNamingStrategy

### ISSUE THREAD

**Issue by aguibert** [eclipse-ee4j/yasson#153]
The default ProeprtyNamingStrategy is `IDENTITY` which means no translation should be performed on the property names.

However, in `ClassModel.equalsReadName()`, a case-insensitive match is done on the read name:
```java
    /**
     * Check if name is equal according to property strategy. In case of {@link CaseInsensitiveStrategy} ignore case.
     * User can provide own strategy implementation, cast to custom interface is not an option.
     *
     * @return True if names are equal.
     */
    private boolean equalsReadName(String jsonName, PropertyModel propertyModel) {
        final String propertyReadName = propertyModel.getReadName();
        if (propertyNamingStrategy instanceof CaseInsensitiveStrategy) {
            return jsonName.equalsIgnoreCase(propertyReadName);
        }
        return jsonName.equalsIgnoreCase(propertyReadName);
    }
```

To properly adhere to IDENTITY naming strategy, we should be performing a case-sensitive equality comparison. 

For example, if we have the following data:

```
data1 = { "SKUData" : "string1" }
data2 = { "skudata" : "string2" }
```

Only `data1` should successfully bind to the `SKUArray.SKUData` field:
```java
public class SkuArray {
    public String SKUData = "";
}
```
however, currently `data2` also binds into the `SKUArray.SKUData` field, which should only happen if CASE_INSENSITIVE naming strategy is specified.

Originally reported by @nadiramra in https://github.com/OpenLiberty/open-liberty/issues/4371

**Comment by aguibert** [comment]
Hi @Verdent, I see you've assigned this issue to yourself but I've already proposed a fix in PR #154.  Could you please review that PR when you get some time?

**Comment by aguibert** [comment]
@bravehorsie or @m0mus could you please have a look at this issue and see if the fix I proposed in PR #154 is sufficient?
